pandas/core/groupby.py

```diff
@@ -2669,7 +2680,7 @@ def _wrap_transformed_output(self, output, names=None):
     def _wrap_applied_output(self, keys, values, not_indexed_same=False):
         if len(keys) == 0:
             # GH #6265
-            return Series([], name=self.name)
+            return Series([], name=self.name, index=keys)
```

Comment:
This works for `Series`. Will add an issue for `DataFrame`, which is materially more complicated